### PR TITLE
Enabling volumeMount.mountPropagation

### PR DESCRIPTION
**Pull request by simonchow765**

### CODE REVIEW

**Comment by simonchow765** [comment]
cc: @bukzor

**Comment by alexsomesan** [comment]
Thanks for your contribution, @simonchow765 !

Would you mind also adding this new attribute to some relevant acceptance tests so that it may be exercised during testing?

**Comment by simonchow765** [comment]
Hey @alexsomesan,
Just FYI I don't plan writing tests for this change any time soon due to other commitments.

**Comment by bukzor** [comment]
:+1:

This is necessary for the Filestore CSI driver to fit under terraform.
https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/

Do you have a similar PR that you can show, to give an idea of the tests you want?

File: kubernetes/schema_container.go [hunk 1]
```diff
@@ -191,6 +191,12 @@ func volumeMountFields() map[string]*schema.Schema {
 			Optional:    true,
 			Description: `Path within the volume from which the container's volume should be mounted. Defaults to "" (volume's root).`,
 		},
+		"mount_propagation": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "None",
```
Comment (alexsomesan) [inline kubernetes/schema_container.go]: There can only be three precise values here, so let's also add a simple validation.

```suggestion
			Default:     "None",
			ValidateFunc: validation.StringInSlice([]string{"None", "HostToContainer", "Bidirectional"}, false),
```

File: kubernetes/structures_container.go [hunk 2]
```diff
@@ -714,6 +717,10 @@ func expandContainerVolumeMounts(in []interface{}) ([]v1.VolumeMount, error) {
 		if subPath, ok := p["sub_path"]; ok {
 			vmp[i].SubPath = subPath.(string)
 		}
+		if mountPropagation, ok := p["mount_propagation"]; ok {
+			temp := v1.MountPropagationMode(mountPropagation.(string))
```
Comment (alexsomesan) [inline kubernetes/structures_container.go]:
```suggestion
			mp := v1.MountPropagationMode(mountPropagation.(string))
```
It's preferable not to have generic variables names like `temp`. It helps at least when grep-ing the code.

File: kubernetes/structures_container.go [hunk 3]
```diff
@@ -714,6 +717,10 @@ func expandContainerVolumeMounts(in []interface{}) ([]v1.VolumeMount, error) {
...
+			vmp[i].MountPropagation = &temp
```
Comment (alexsomesan) [inline kubernetes/structures_container.go]:
```suggestion
			vmp[i].MountPropagation = &mp
```
See above comment.

**Comment by JordanP** [comment]
I didn't see this PR before so I went with my own: https://github.com/terraform-providers/terraform-provider-kubernetes/pull/552

**Comment by aareet** [comment]
Closing this since #552 was merged

**Pull request closed without merging**